src/charm.py

```diff
@@ -205,6 +207,110 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
+            kf_profiles = self._check_kf_profiles(interfaces)
+            self.handle_ingress(interfaces)
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        try:
+            self.unit.status = MaintenanceStatus("Creating k8s resources")
+            self.k8s_resource_handler.apply()
+        except ApiError:
+            self.logger.error(traceback.format_exc())
+            self.unit.status = BlockedStatus("kubernetes resource creation failed")
+            return
+        self.handle_ingress(interfaces)
+        kf_profiles = list(kf_profiles.get_data().values())[0]
+        self.profiles_service = kf_profiles["service-name"]
+        try:
+            self._update_layer()
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        self.model.unit.status = ActiveStatus()
+
+    def _get_sidebar_config(self) -> Dict:
+        try:
+            current_configmap = self.k8s_resource_handler.lightkube_client.get(
```

Comment:
I hadn't thought about using the `k8s_resource_handler`'s `lightkube_client` here.  If you did this because you like it then :+1:, if you did it because you thought we'd frown on creating your own local lightkube client as well then feel free to make a lightkube_client locally as well

Reply:
so the `lightkube_client` of `k8s_resource_handler's` is not private so I hoped it should be used like that :D